docs/diagnostics/PX1016.md

```diff
@@ -0,0 +1,36 @@
+# PX1016
+This document describes the PX1016 diagnostic.
+
+## Summary
+
+| Code   | Short Description                                                                                                                      | Type                           | Code Fix    | 
+| ------ | -------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------ | ----------- | 
+| PX1016 | A DAC or graph extension must include the public static `IsActive` method with no parameters and the `bool` return type. Extensions that are constantly active reduce performance. Suppress the error if you need the DAC extension to be constantly active. | Error | Unavailable | 
+
+## Diagnostic Description
+A DAC or a graph extension must include the public static `IsActive` method with no parameters and the `bool` return type. Extensions that are constantly active reduce performance.
+
+To fix the issue, in the `IsActive` method, you should enable the DAC or graph extension if the feature for which this extension is needed is enabled. 
+
+In case you do not want to define the `IsActive` method as suggested above, you should suppress the diagnostic for better performance, otherwise the `IsActive` method will be called constantly to check whether the extension is enabled.
+
+## Example of Code that Results in the Error
+
+```C#
+// A diagnostic must be shown on SOOrderEntryExt 
+public class SOOrderEntryExt : PXGraphExtension<SOOrderEntry>
+{
+    ...
+	// no isActive method
+}
+```
+
+## Example of the Fixed Code
```

Comment:
Please also add an example with a suppression comment:

```
// Acuminator disable once PX1016 ExtensionDoesNotDeclareIsActiveMethod because this extension should be always enabled
public class SOOrderEntryExt : PXGraphExtension<SOOrderEntry>
{
    ...
}
```